Content.Client/PDA/Ringer/RingtoneMenu.xaml.cs

```diff
@@ -0,0 +1,129 @@
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Maths;
+using System;
+using Content.Client;
+
+namespace Content.Client.PDA.Ringer
+{
+    [GenerateTypedNameReferences]
+    public partial class RingtoneMenu : SS14Window
+    {
+        private string[]? _previousNoteInputs = new string[4];
+        public enum Note
+        {
+            A,
+            Asharp,
+            B,
+            C,
+            Csharp,
+            D,
+            Dsharp,
+            E,
+            F,
+            Fsharp,
+            G,
+            Gsharp
+        }
+
+        public RingtoneMenu()
+        {
+            RobustXamlLoader.Load(this);
+
+            //RingerNoteOneInput
+            RingerNoteOneInput.OnTextChanged += _ => //Prevents unauthorized characters from being entered into the LineEdit
+            {
+                RingerNoteOneInput.Text = RingerNoteOneInput.Text.ToUpper();
+
+                if(!isNote(RingerNoteOneInput.Text))
+                {
+                    RingerNoteOneInput.Text = _previousNoteInputs[0];
+                }
+                else
+                {
+                    _previousNoteInputs[0] = RingerNoteOneInput.Text;
+                }
+
+                RingerNoteOneInput.CursorPosition = RingerNoteOneInput.Text.Length; //Resets caret position to the end of the typed input
+            };
```

Comment:
For localisation reasons this should ideally be something like a dropdown or else ringers won't work.